sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py

```diff
@@ -133,6 +144,14 @@ def resolve(self) -> None:
         base_path = self._additional_includes_file_path.parent
         for additional_include in self._includes:
             src_path = (base_path / additional_include).resolve()
+            if self._zip_need_to_compress(src_path):
+                zip_additional_include = (base_path / additional_include).resolve()
+                folder_to_zip = zip_additional_include.parent / zip_additional_include.stem
+                with tempfile.TemporaryDirectory() as tmp_dir:
+                    tmp_folder_to_zip = (Path(tmp_dir) / "zip").resolve()
+                    shutil.copytree(folder_to_zip, tmp_folder_to_zip, ignore=shutil.ignore_patterns("__pycache__"))
```

Comment:
maybe extract this so that ignore logic is shared

Reply:
extract this to a function `_copy_folder_ignore_pycache` in `sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py`